src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,1055 @@
+import numpy as np
+
+from qibo import gates
+from qibo.backends import GlobalBackend
+from qibo.models import Circuit
+
+
+def prepare_states(k, nqubits=None):
+    if nqubits is not None:
+        if nqubits == 1:
+            k_to_gates_1qb = {
+                0: (),
+                1: (gates.X(0),),
+                2: (gates.H(0),),
+                3: (gates.H(0), gates.S(0)),
+            }
+
+            circ = Circuit(nqubits, density_matrix=True)
+            for gate in k_to_gates_1qb[k]:
+                circ.add(gate)
+
+            return circ
+
+        elif nqubits == 2:
+            k_to_gates_2qb = {
+                0: (),
+                1: (gates.X(1),),
+                2: (gates.H(1),),
+                3: (gates.H(1), gates.S(1)),
+                4: (gates.X(0),),
+                5: (
+                    gates.X(0),
+                    gates.X(1),
+                ),
+                6: (
+                    gates.X(0),
+                    gates.H(1),
+                ),
+                7: (gates.X(0), gates.H(1), gates.S(1)),
+                8: (gates.H(0),),
+                9: (
+                    gates.H(0),
+                    gates.X(1),
+                ),
+                10: (
+                    gates.H(0),
+                    gates.H(1),
+                ),
+                11: (gates.H(0), gates.H(1), gates.S(1)),
+                12: (
+                    gates.H(0),
+                    gates.S(0),
+                ),
+                13: (
+                    gates.H(0),
+                    gates.S(0),
+                    gates.X(1),
+                ),
+                14: (
+                    gates.H(0),
+                    gates.S(0),
+                    gates.H(1),
+                ),
+                15: (gates.H(0), gates.S(0), gates.H(1), gates.S(1)),
+            }
+
+            circ = Circuit(nqubits, density_matrix=True)
+            for gate in k_to_gates_2qb[k]:
+                circ.add(gate)
+
+            return circ
+
+        else:
+            raise ValueError(
+                f"nqubits given as {nqubits}. nqubits needs to be either 1 or 2."
+            )
+
+    else:
+        raise ValueError(
+            f"nqubits given as {nqubits}. nqubits needs to be either 1 or 2."
+        )
```

Comment:
```suggestion
def prepare_states(k, nqubits):
    if nqubits != 1 and nqubits != 2:
        raise ValueError(
            f"nqubits given as {nqubits}. nqubits needs to be either 1 or 2."
        )

    gates_list=[(gates.I,),(gates.X,),(gates.H,),(gates.H,gates.S)]

    k_to_gates = list(product(gates_list,repeat=nqubits))[k]
    circ = Circuit(nqubits, density_matrix=True)    
    for q in range(len(k_to_gates)):
        gate_q = [gate(q) for gate in k_to_gates[q]]
        circ.add(gate_q)
        
    return circ
```

By using `itertools.product`, we can avoid hardcoding the gates to prepare the single qubit fiducial states `|0>`, `|1>`, `|+>`, `|->` , as well as all tensor product combinations to generate the two qubit states.